python/ql/src/experimental/dataflow/internal/DataFlowPublic.qll

```diff
@@ -304,14 +322,23 @@ class DictionaryElementContent extends TDictionaryElementContent, Content {
 
   DictionaryElementContent() { this = TDictionaryElementContent(key) }
 
-  /** Gets the index for this tuple element */
+  /** Gets the key for this dictionary element. */
   string getKey() { result = key }
 
-  /** Gets a textual representation of this element. */
   override string toString() { result = "Dictionary element at key " + key }
 }
 
 class DictionaryElementAnyContent extends TDictionaryElementAnyContent, Content {
-  /** Gets a textual representation of this element. */
   override string toString() { result = "Any dictionary element" }
 }
+
+class AttributeContent extends TAttributeContent, Content {
+  private string attr;
```

Comment:
What role does the `private` annotation play here? (I'm mostly asking for my own edification. I've never had a use for private fields, so I'm curious as to when they are appropriate to use.)

Reply:
I think it means that you can only get to it via `getAttr` from outside. I do not remember why we have it here, so chances are it is copied from [a similar place in C#](https://github.com/github/codeql/blob/a93a84fb2e75d5181094260e9cc7d7cffca90ae0/csharp/ql/src/semmle/code/csharp/dataflow/internal/DataFlowPublic.qll#L232-L233).